### PR TITLE
Fix userguide save process customisation example

### PR DESCRIPTION
**Pull request by lbdreyer**
I noticed a slight error in the "Customising the save process" code example. This fixes ` message` to be `grid_message`

Whilst I was on the page, I also noticed something (admittedly very minor) that I don't like in the first code example block:
```python
import iris
filename = iris.sample_data_path('uk_hires.pp')
cubes = iris.load(filename)
cube = cubes[0]
iris.save(cubes, '/tmp/uk_hires.nc')
```
I don't like the fact that a cube is indexed from the cubelist, but then the next line is saving the original cubelist. Someone would never do this so it shouldn't be example code.

I understand that the indexing is done as, further down, there are examples that use `cube` to demonstrate that a single cube can also be saved. Instead, I propose that the indexing should be done at the point of saving or perhaps just do the indexing at a later point (e.g. in the 'Controlling the save process' example code.

The other option would be to save the single cube, but then save it to a different filename, i.e.
```python
import iris
filename = iris.sample_data_path('uk_hires.pp')
cubes = iris.load(filename)
cube = cubes[0]
iris.save(cube, '/tmp/new_file.nc')
```

As I said, this is minor, and I am (sort of) happy to remove it from this PR.

### CODE REVIEW

File: docs/iris/src/userguide/saving_iris_cubes.rst [hunk 1]
```diff
@@ -77,8 +76,8 @@ For example, a GRIB2 message with a particular known long_name may need to be sa
 		    gribapi.grib_set_long(grib_message, "typeOfStatisticalProcess", 1)
```
Comment (ajdawson) [inline docs/iris/src/userguide/saving_iris_cubes.rst]: The indentation looks off here too.

File: docs/iris/src/userguide/saving_iris_cubes.rst [hunk 2]
```diff
@@ -46,12 +45,12 @@ Controlling the save process
...
-    >>> iris.save(cube, "myfile.pp")
+    >>> iris.save(cubes[0], "myfile.pp")
     >>> # Save a cube list to a PP file, appending to the contents of the file
     >>> # if it already exists
     >>> iris.save(cubes, "myfile.pp", append=True)
     >>> # Save a cube to netCDF, defaults to NETCDF4 file format
-    >>> iris.save(cube, "myfile.nc")
+    >>> iris.save(cubes[0], "myfile.nc")
     >>> # Save a cube list to netCDF, using the NETCDF4_CLASSIC storage option
     >>> iris.save(cubes, "myfile.nc", netcdf_format="NETCDF3_CLASSIC")
```
Comment (rhattersley) [inline docs/iris/src/userguide/saving_iris_cubes.rst]: The version numbers of the netCDF format in the comment and the code are inconsistent. I don't think it matters which we go with.

Comment (rhattersley) [inline docs/iris/src/userguide/saving_iris_cubes.rst]: Other than this I'm :+1: (Sorry for the scope creep! :unamused: )

Comment (lbdreyer) [inline docs/iris/src/userguide/saving_iris_cubes.rst]: Do you have a preference for NETCDF4_CLASSIC vs NETCDF3_CLASSIC?

Comment (rhattersley) [inline docs/iris/src/userguide/saving_iris_cubes.rst]: +0 for NETCDF3_CLASSIC ... it's slightly more obvious at a glance that it is different from the default.

**Comment by pelson** [comment]
:+1:

File: lib/iris/io/__init__.py [hunk 3]
```diff
@@ -362,7 +362,7 @@ def save(source, target, saver=None, **kwargs):
         # Save a cube to netCDF, defaults to NETCDF4 file format
         iris.save(my_cube, "myfile.nc")
 
-        # Save a cube list to netCDF, using the NETCDF4_CLASSIC storage option
+        # Save a cube list to netCDF, using the NETCDF3_CLASSIC storage option
```
Comment (rhattersley) [inline lib/iris/io/__init__.py]: Well spotted :eyes: :+1:

**Comment by rhattersley** [comment]
Thanks @lbdreyer :smile:
